build.sbt

```diff
@@ -19,6 +19,12 @@ lazy val LanguageVersion = LanguageVersions.head
 // ==========================================
 
 sharedSettings
+version.in(ThisBuild) ~= { old =>
+  val suffix =
+    if (sys.props.contains("scalameta.snapshot")) "-SNAPSHOT"
+    else ""
+  customVersion.getOrElse(old.replace('+', '-') + suffix)
+}
```

Comment:
Are these the usual SNAPSHOT releases, which make sbt do updates like crazy, or there's something special about them, since we seem to be publishing them to Sonatype releases?

Reply:
it's possible to configure sbt to treat -SNAPSHOT versions like normal versions, we never overwrite an old release because every merge has a unique commit that is included in the version number.